### PR TITLE
Add Proverb

### PR DESCRIPTION
**Pull request by NadiaJalalTamimi**
Make the add proverbs modal in the dashboard router.

Screenshot (148)

### CODE REVIEW

**Review by reemsh2000 — APPROVED** [review]

**Review by modrayyes — APPROVED** [review]

File: client/src/components/Proverbs/Addproverbs.jsx [hunk 1]
```diff
@@ -0,0 +1,62 @@
+import React, { useState } from 'react';
+import {
+  Modal, Button, Input, message,
+} from 'antd';
+import './proverb.css';
+import http from '../../services/httpService';
+
+function Addproverbs() {
+  const [visible, setVisible] = useState(false);
+  const [value, setValue] = useState({ proverb: '' });
```
Comment (Nizar7zak) [inline client/src/components/Proverbs/Addproverbs.jsx:10]: I guess it's better if you set an initial value here like this (' ')

File: client/src/components/Proverbs/Addproverbs.jsx [hunk 2]
```diff
@@ -0,0 +1,62 @@
...
+      if (value.proverb !== '') {
+        const proverb = await http.post('api/v1/proverb', value);
+        message.success(proverb.response.data.message);
+      } else {
+        setVisible(true);
+        throw new Error('يجب عليك إضافة مثل');
+      }
+    } catch (error) {
+      message.error(error.response.data.message);
+      console.log(error);
```
Comment (Nizar7zak) [inline client/src/components/Proverbs/Addproverbs.jsx:25]: you should avoid any console line in your code. I guess you forget remove it.

File: client/src/components/Proverbs/Addproverbs.jsx [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+        title="أضف المثل"
+        centered
+        visible={visible}
+        onOk={() => setVisible(false)}
+        onCancel={() => setVisible(false)}
+        width={1000}
+        footer={null}
+      >
+        <div className="proverbcontainer">
+          <div><Input type="text" onChange={handleChange} className="input" placeholder="أضف هنا المثال" required /></div>
```
Comment (Nizar7zak) [inline client/src/components/Proverbs/Addproverbs.jsx:47]: why you nested an input component with div? you can style an input it took class name

File: client/src/components/Proverbs/Addproverbs.jsx [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+          <div className="btns">
+            <Button size="large" type="primary" onClick={handleOpen} style={{ backgroundColor: 'rgb(24, 85, 3)', borderRadius: '8px', padding: '0px 40px' }}>
```
Comment (Nizar7zak) [inline client/src/components/Proverbs/Addproverbs.jsx:49]: here you should use our common button component,

File: client/src/components/Proverbs/proverb.css [hunk 5]
```diff
@@ -0,0 +1,20 @@
...
+   width:35%;
+    display: flex;
+    flex-direction: row;
+    justify-content:space-around;
+    align-self: center;
+
+}
+.proverbcontainer{
+    width:100%;
+    height: 270px;
```
Comment (Nizar7zak) [inline client/src/components/Proverbs/proverb.css:16]: avoid px value, you should set an relative value like vh.

**Review by Nizar7zak — CHANGES_REQUESTED** [review]

File: client/src/components/Proverbs/Addproverbs.jsx [hunk 6]
```diff
@@ -0,0 +1,62 @@
...
+  console.log(value);
+
+  const sendProverb = async () => {
+    try {
+      if (value.proverb !== '') {
+        const proverb = await http.post('api/v1/proverb', value);
+        message.success(proverb.response.data.message);
+      } else {
+        setVisible(true);
+        throw new Error('يجب عليك إضافة مثل');
```
Comment (LinaYahya) [inline client/src/components/Proverbs/Addproverbs.jsx:21]: why to throw instead use messsgae.warn here,
OR I prefer to validate input using Form.Item

File: client/src/components/Proverbs/Addproverbs.jsx [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+  };
+  const handleOpen = () => {
+    setVisible(false);
+    sendProverb();
+  };
+  return (
+    <>
+      <Button type="secondary" onClick={() => setVisible(true)}>
+        إضافة مثل
+      </Button>
```
Comment (LinaYahya) [inline client/src/components/Proverbs/Addproverbs.jsx:36]: remove this button and pass visible and setVisible as a props

File: client/src/components/Proverbs/Addproverbs.jsx [hunk 8]
```diff
@@ -0,0 +1,62 @@
...
+      <Modal
+        title="أضف المثل"
+        centered
+        visible={visible}
+        onOk={() => setVisible(false)}
+        onCancel={() => setVisible(false)}
+        width={1000}
+        footer={null}
+      >
+        <div className="proverbcontainer">
```
Comment (LinaYahya) [inline client/src/components/Proverbs/Addproverbs.jsx:46]: Check how to add form to modal we'll use antd Form instead

**Review by LinaYahya — CHANGES_REQUESTED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by modrayyes — APPROVED** [review]

**Review by LinaYahya — APPROVED** [review]

**Review by Nizar7zak — APPROVED** [review]

**Pull request merged**